### PR TITLE
Move non-root-enabler into operator image

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now move the non-root-enabler logic into an own internal
`nonrootenabler` package and re-use it within the operator binary.

This allows mocked unit testing as well as moving away from another
container image (docker.io/bash) dependency.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
Yes

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Removed `docker.io/bash:5` container image dependency for non-root-enabler logic.
```

### CODE REVIEW

File: internal/pkg/config/config.go [hunk 1]
```diff
@@ -25,6 +25,12 @@ const (
 	// OperatorName is the name when referring to the operator.
 	OperatorName = "security-profiles-operator"
 
+	// OperatorRoot is the root directory of the operator.
+	OperatorRoot = "/var/lib/security-profiles-operator"
+
+	// UserRootless is the user which runs the operator.
+	UserRootless = 65535
```
Comment (JAORMX) [inline internal/pkg/config/config.go:32]: why not make this int64 already? that way you don't need to cast it later on

File: internal/pkg/controllers/spod/bindata/spod.go [hunk 2]
```diff
@@ -67,36 +66,16 @@ var Manifest = &appsv1.DaemonSet{
...
-						if [ ! -L ` + config.ProfilesRootPath + ` ]; then
-							/bin/ln -s ` + operatorRoot + ` ` + config.ProfilesRootPath + `
-						fi
-
-						/bin/chown -R 65535:65535 ` + operatorRoot + `
-						cp -f -v /opt/seccomp-profiles/* ` + config.KubeletSeccompRootPath + `
-					`},
+						Name:            "non-root-enabler",
+						Args:            []string{"non-root-enabler"},
+						ImagePullPolicy: v1.PullNever,
```
Comment (JAORMX) [inline internal/pkg/controllers/spod/bindata/spod.go:71]: The daemon container defaults to `PullAlways`. Let's make both containers default to the same pull policy instead.

File: internal/pkg/config/config.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/config/config.go:32]: `os.Chown` accepts only `int`: https://golang.org/pkg/os/#Chown

This is used within nonrootenabler.go, so I would have a need for casting there.

File: internal/pkg/controllers/spod/bindata/spod.go (hunk 2 above)
Comment (saschagrunert) [reply internal/pkg/controllers/spod/bindata/spod.go:71]: Sure, fixing as suggested :+1:

File: internal/pkg/config/config.go (hunk 1 above)
Comment (JAORMX) [reply internal/pkg/config/config.go:32]: gotcha. That's fine then.

**Comment by saschagrunert** [comment]
Had to rebase once again. I think the main issue was the missing `Chmod` in case the operator root already existed. This should be fixed now, too.

**Pull request merged**